seisbench/models/base.py

```diff
@@ -1519,273 +1478,363 @@ def _cut_fragments_point(self, t0, block, stations, argdict):
+        pred_time = meta.start_time + meta.pred_sample[0] / argdict["sampling_rate"]
+        pred_rate = argdict["sampling_rate"] * pred_sample_factor
 
-            pred_time = t0 + pred_sample[0] / argdict["sampling_rate"]
-            pred_rate = argdict["sampling_rate"] * prediction_sample_factor
+        # Maximum number of predictions covering a point
+        max_coverage = int(np.ceil(meta.in_samples / (meta.in_samples - overlap) + 1))
 
-            output = ((pred_rate, pred_time, preds), stations)
+        pred_length = int(
+            np.ceil((np.max(offsets) + meta.in_samples) * pred_sample_factor)
+        )
+        pred_merge = np.full_like(
+            segments[0],
+            np.nan,
+            shape=(
+                max_coverage,
+                segments[0].shape[0],
+                pred_length,
+                segments[0].shape[-1],
+            ),
+        )
 
-            del buffer[key]
+        for i, (data, start) in enumerate(zip(segments, offsets)):
+            pred_start = int(start * pred_sample_factor)
+            pred_merge[
+                i % max_coverage, :, pred_start : pred_start + data.shape[1], :
+            ] = data
+
+        with warnings.catch_warnings():
+            if stack_method == "avg":
+                warnings.filterwarnings(action="ignore", message="Mean of empty slice")
+                preds = bn.nanmean(pred_merge, axis=0)
+            elif stack_method == "max":
+                warnings.filterwarnings(action="ignore", message="All-NaN")
+                preds = bn.nanmax(pred_merge, axis=0)
+            # Case of stack_method not in avg or max is caught by assert above
+
+        # preds = np.squeeze(preds)
+        if self._grouping.grouping == "channel":
+            preds = preds[0, :, 0]
+        elif self._grouping.grouping == "instrument":
+            preds = preds[0]
 
-        return output
+        return PredictionsStacked(
+            data=preds,
+            stations=meta.stations,
+            sampling_rate=pred_rate,
+            start_time=pred_time,
+        )
 
-    def _predict_buffer(self, buffer, argdict):
+    def _predict_windows(
+        self,
+        segments: list[TraceSegment],
+        argdict: dict,
+    ) -> list[PredictionSegment]:
         """
         Batches model inputs, runs preprocess, prediction and postprocess, and unbatches output
 
         :param buffer: List of inputs to the model
         :return: Unpacked predictions
         """
+        data = [window.data for window in segments]
         if self.allow_padding:
-            fragments = seisbench.util.pad_packed_sequence(buffer)
+            fragments = seisbench.util.pad_packed_sequence(data)
         else:
-            fragments = np.stack(buffer, axis=0)
-        fragments = torch.tensor(fragments, device=self.device, dtype=torch.float32)
+            fragments = np.array(data)
+        fragments = torch.from_numpy(fragments).to(self.device).float()
```

Comment:
I'd use `torch.to_tensor(fragments, device=self.device, dtype=torch.float32)`. This will automatically minimize the number of copies.